lti_consumer/plugin/views.py

```diff
@@ -54,3 +58,22 @@ def launch_gate_endpoint(request, suffix):
         )
     except:  # pylint: disable=bare-except
         return HttpResponse(status=404)
+
+
+@csrf_exempt
+@require_http_methods(["POST"])
+def access_token_endpoint(request, usage_id=None):
+    """
+    Gate endpoint to enable tools to retrieve access tokens
+    """
+    try:
+        usage_key = UsageKey.from_string(usage_id)
+
+        return handle_xblock_callback_noauth(
+            request=request,
+            course_id=str(usage_key.course_key),
+            usage_id=str(usage_key),
+            handler='lti_access_token'
+        )
+    except:  # pylint: disable=bare-except
```

Comment:
It's a good idea to at least log this exception, since we are returning 404 for any problem here.

Reply:
Improved exception handling here. 
I've tried to address all applicable errors from https://tools.ietf.org/html/rfc6749#section-5.2

Reply:
It works the same way as the other methods. 
But I can try to add a logger :)